EventSourcing.Core/Records/RecordTypesCache.cs

```diff
@@ -0,0 +1,57 @@
+using System.Reflection;
+
+namespace EventSourcing.Core.Types;
+
+public sealed class RecordTypesCache
+{
+    // Static RecordTypes cache
+    private static readonly List<Type> AssemblyRecordTypes = AppDomain.CurrentDomain
+        .GetAssemblies()
+        .SelectMany(x => x.GetTypes())
+        .Where(type => typeof(Record).IsAssignableFrom(type) && type.IsClass && !type.IsAbstract && type.IsPublic)
+        .ToList();
+    private static readonly Dictionary<Type, string> RecordTypesRev;
+    static RecordTypesCache()
+    {
+        // Create dictionaries mapping from Record.Type to Record Type string
+        RecordTypesRev = AssemblyRecordTypes.ToDictionary(type => type, type => type.GetCustomAttribute<RecordType>()?.Value ?? type.Name);
```

Comment:
The ```RecordTypesRev``` definition and initialisation can be merged. The static Constructor will then not be needed :)